dev/tools/controllerbuilder/scaffold/controller.go

```diff
@@ -42,30 +41,33 @@ var funcMap = template.FuncMap{
 	"ToLower": strings.ToLower,
 }
 
-func RegisterController(service, kind string) error {
-	// Read register file
-	directControllerPkgPath, err := buildDirectControllerPath()
-	if err != nil {
-		return nil
+func NewControllerBuilder(rootPath, service, proto string) *ControllerBuilder {
+	return &ControllerBuilder{
+		rootPath: rootPath,
+		service:  service,
+		proto:    proto,
 	}
-	registerFilePath := filepath.Join(directControllerPkgPath, "register", "register.go")
+}
+
+type ControllerBuilder struct {
+	rootPath string
+	service  string
+	proto    string
+}
+
+func (c *ControllerBuilder) RegisterController() error {
+	// Read register file
+	registerFilePath := filepath.Join(c.getDirectPath(), "register", "register.go")
 	fset := token.NewFileSet()
 	f, err := parser.ParseFile(fset, registerFilePath, nil, parser.ParseComments)
 	if err != nil {
 		return err
 	}
 
-	// Get main model name
-	bi, ok := debug.ReadBuildInfo()
-	if !ok {
-		return fmt.Errorf("could not read build info")
-	}
-	modelPath := strings.TrimSuffix(bi.Main.Path, currRelPath)
-
-	importPath := filepath.Join(modelPath, directControllerRelPath, service)
+	importPath := filepath.Join("github.com/GoogleCloudPlatform/k8s-config-connector", directControllerRelPath, c.service)
```

Comment:
I used a fixed import path because the debug.ReadBuildInfo cannot find the module if running in a different git repo.

Reply:
sgtm.  You can always make it a flag in future (and pass down the Options struct)